sdk/test-utils/perf/src/utils/profiling.ts

```diff
@@ -0,0 +1,45 @@
+import { Session } from "node:inspector";
+import * as fs from "fs-extra";
+
+export async function runWithCpuProfile(
+  functionToProfile: () => Promise<void>,
+  profileFilePath: string | undefined
+) {
+  const session = new Session();
+  session.connect();
+  session.post("Profiler.enable", () => {
```

Comment:
How compatible and stable is this API? Works on Node 16 & 18?

Reply:
It always existed(available in 10.x too), works in 16 and 18, and even 19.
In 19, they added the version with [promises](https://nodejs.org/docs/latest-v19.x/api/all.html#all_inspector_cpu-profiler).